### ISSUE TITLE
New check: STAT tables in static fonts

### ISSUE THREAD

**Issue by simoncozens** [googlefonts/fontbakery#4149]
## What needs to be checked?

Static instances need different stuff in their STAT tables to variable fonts. Fontbakery should tell us if a "variable" STAT table has ended up in a static instance.

## Detailed description of the problem 

Adobe feature syntax allows for the definition of a STAT table. Fonts built with a hand-coded STAT table in feature syntax may be built either as static or variable, but will end up with the same STAT table. This is a problem, because a STAT table which works on variable fonts will not be appropriate for static instances. The [examples](https://learn.microsoft.com/en-gb/typography/opentype/spec/stat#examples) in the OT spec of non-variable fonts with a STAT table show that the table entries must be restricted to those entries which refer to the static font's position in the designspace. i.e. a Regular weight static should *only* have the following entry for the weight axis:

```
        <AxisIndex value="0"/>
        <Flags value="2"/>  <!-- ElidableAxisValueName -->
        <ValueNameID value="265"/>  <!-- Regular -->
        <Value value="400.0"/>
```

However, if the STAT table intended for a variable font is compiled into a static, it will have many entries for this axis. In this case, Windows will read the *first* entry only, causing all instances to report themselves as "Thin Condensed".

## Optional fix

Just drop the STAT table from the static.

## Resources and exact process needed to replicate

https://github.com/notofonts/latin-greek-cyrillic/issues/429

## Expected Profile

- [x] Universal (broadly accepted best practices on the type design community)

This'll mess you up on Windows if you get it wrong.

## Expected Result

Fontbakery should:

* WARN if a non-variable font has a STAT table at all, since they're a pretty rare thing and you're almost certainly better off without them (just rely on WWS/style linking).
* FAIL if a non-variable font has more than one AxisValue per axis.

## Severity assessment

4

**Comment by felipesanches** [comment]
I'm starting to work on this one now.

**Comment by felipesanches** [comment]
The new check was implemented at PR #4151 not including the suggested **WARN** on the mere presence of a STAT table on static fonts because @Lorp suggested in a conversation that having the STAT table would actually be desirable. Also @twardoch presented some ideas about this topic on a private conversation.

I am closing this issue, and encourage others to open a separate one in case further changes (like the proprosed WARN) are still needed.
